src/python_testing/TC_EEVSE_Utils.py

```diff
@@ -0,0 +1,190 @@
+        asserts.assert_equal(res.Header.EventId, expected_event.event_id, "Expected event ID not found in event report")
+        return res.Data
+
+
+class EEVSEBaseTestHelper:
+
+    async def read_evse_attribute_expect_success(self, endpoint, attribute):
+        full_attr = getattr(Clusters.EnergyEvse.Attributes, attribute)
+        cluster = Clusters.Objects.EnergyEvse
+        return await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=full_attr)
+
+    async def check_evse_attribute(self, attribute, expected_value, endpoint: int = 1):
+        value = await self.read_evse_attribute_expect_success(endpoint=endpoint, attribute=attribute)
+        asserts.assert_equal(value, expected_value,
+                             f"Unexpected '{attribute}' value - expected {expected_value}, was {value}")
+
+    async def get_supported_energy_evse_attributes(self, endpoint):
+        return await self.read_evse_attribute_expect_success(endpoint, "AttributeList")
+
+    async def send_enable_charge_command(self, endpoint: int = 1, charge_until: int = None, timedRequestTimeoutMs: int = 3000,
+                                         min_charge: int = None, max_charge: int = None, expected_status: Status = Status.Success):
+        try:
+            await self.send_single_cmd(cmd=Clusters.EnergyEvse.Commands.EnableCharging(
+                chargingEnabledUntil=charge_until,
+                minimumChargeCurrent=min_charge,
+                maximumChargeCurrent=max_charge),
+                endpoint=endpoint,
+                timedRequestTimeoutMs=timedRequestTimeoutMs)
+
+        except InteractionModelError as e:
+            asserts.assert_equal(e.status, expected_status, "Unexpected error returned")
+
+    async def send_disable_command(self, endpoint: int = 1, timedRequestTimeoutMs: int = 3000, expected_status: Status = Status.Success):
+        try:
+            await self.send_single_cmd(cmd=Clusters.EnergyEvse.Commands.Disable(),
+                                       endpoint=endpoint,
+                                       timedRequestTimeoutMs=timedRequestTimeoutMs)
+
+        except InteractionModelError as e:
+            asserts.assert_equal(e.status, expected_status, "Unexpected error returned")
+
+    async def send_start_diagnostics_command(self, endpoint: int = 1, timedRequestTimeoutMs: int = 3000,
+                                             expected_status: Status = Status.Success):
+        try:
+            await self.send_single_cmd(cmd=Clusters.EnergyEvse.Commands.StartDiagnostics(),
+                                       endpoint=endpoint,
+                                       timedRequestTimeoutMs=timedRequestTimeoutMs)
+
+        except InteractionModelError as e:
+            asserts.assert_equal(e.status, expected_status, "Unexpected error returned")
+
+    async def send_test_event_triggers(self, enableKey=bytes([b for b in range(16)]), eventTrigger=0x0099000000000000):
```

Comment:
Is this a hardcoded enable key?

Reply:
It is currently, I was looking for a way in matter_testing_support to be able to get this value but for now it is.

is there something in the framework to get this?

similarly the endpoint id is hardcoded as 1 for now.

Reply:
Is config.global_test_params the right place for it? Or should this come from a PICS file? (or both)?

It could come from a `--hex-arg enableKey==000102030405060708090A0B0C0D0E0F` ?

I managed to get this to work, but not sure if we should add something more specific to matter_testing_support.py as an explicit argument?


```    async def send_test_event_triggers(self, enableKey: bytes = None, eventTrigger=0x0099000000000000):
        # get the test event enable key or assume the default
        # This can be passed in on command line using 
        #    --hex-arg enableKey:000102030405060708090a0b0c0d0e0f 
        if enableKey is None:
            if 'enableKey' not in self.matter_test_config.global_test_params:
                enableKey = bytes([b for b in range(16)])
            else:
                enableKey = self.matter_test_config.global_test_params['enableKey'])

        try:
            await self.send_single_cmd(endpoint=0,
                                       cmd=Clusters.GeneralDiagnostics.Commands.TestEventTrigger(
                                           enableKey,
                                           eventTrigger)
                                       )

        except InteractionModelError as e:
            asserts.fail(f"Unexpected error returned - {e.status}")
         
```

Reply:
Based on slack conversation I'll commit this as the way to go.